src/cli/exportLicenseRefUsingSPDX.php

```diff
@@ -0,0 +1,255 @@
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along
+ with this program; if not, write to the Free Software Foundation, Inc.,
+ 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ ***********************************************************/
+
+/**
+ * @file exportLicenseRefUsingSPDX.php
+ * @brief Export a licenses from SPDX.
+ *
+ * This is typically used to export license data from SPDX json
+ **/
+
+class exportLicenseRef
+{
+  /**
+   * @var mapArrayData $mapArrayData
+   * actual names for license/exception in SPDX for text and licenseid
+   */
+  private $mapArrayData = array(
+    'licenses' => array('licenseId', 'licenseText', 'name'),
+    'exceptions' => array('licenseExceptionId', 'licenseExceptionText', 'name')
+  );
+
+
+  function startProcessingLicenseData()
+  {
+    global $argv;
+    $updateWithNew = '';
+    $updateExisting = '';
+    $addNewLicense = '';
+    $newLicenseRefData = array();
+    $usage = "Usage: " . basename($argv[0]) . " [options]
+
+      Create new licenseref.json file.  Options are: 
+        -E    Update all existing licenses and also add new licenses.
+              (NOTE: there may be failure of test cases)
+
+        -e    Only update existing licenses.
+              (NOTE: there may be failure of test cases)
+
+        -n    Only add new licenses.
+ 
+       --type Usually licenses/exceptions (optional)
+              (ex: --type 'licenses')
+
+       --url From where you want to download (optional)
+             (ex: --url 'https://spdx.org/licenses/licenses.json')
+
+      Additional note:
+        (if --type and --url is empty then the script will automatically download the from below)
+          For type 'licenses' URL is : $scanList[licenses]
+
+          For type 'exceptions' URL is : $scanList[exceptions]";
+    $scanList = array(
+    'licenses' => 'https://spdx.org/licenses/licenses.json',
+    'exceptions' => 'https://spdx.org/licenses/exceptions.json'
+    );
+
+    $options = getopt("hcEen", array("type:","url:"));
+    /* get type and url if exists if not set them to empty */
+    $type = array_key_exists("type", $options) ? $options["type"] : '';
+    $URL =  array_key_exists("url", $options) ? $options["url"] : '';
+    foreach ($options as $option => $optVal) {
+      switch ($option) {
+        case 'c': /* used by fo_wrapper */
+          break;
+        case 'E': /* Update all existing licenses and also add new licenses */
+          $updateWithNew = $option;
+          break;
+        case 'e': /* Only update existing licenses */
+          $updateExisting = $option;
+          break;
+        case 'n': /* only add new licenses */
+          $addNewLicense = $option;
+          break;
+        case 'h': /* help */
+          $showUsage = true;
+          break;
+      }
+    }
+
+    if ($showUsage) {
```

Comment:
in other programming languages this would be headache, because $showUsage is only defined in case 'h' above. just q a question: this work well in all circumstances?

Reply:
yes it should work in all circumstances.